cookbooks/global_melt.prm

```diff
@@ -0,0 +1,380 @@
+# Cookbook for a global-scale 2D box mantle convection model 
+# with melt migration. 
+# In this file we will go through all of the steps that are  
+# required for adding melting and melt transport to a mantle
+# convection simulation. 
+
+set Dimension                              = 2
+set Adiabatic surface temperature          = 1600
+set Maximum time step                      = 1e6
+set Output directory                       = with_melt
+set Linear solver tolerance                = 1e-8
+set Use years in output instead of seconds = true
+
+# For models with melt migration, there is a nonlinear coupling between
+# the Stokes system, the temperature, and the advection equation for the 
+# porosity (several material properties, such as the viscosities and the
+# permeability depend nonlinearly on the porosity; and changes in temperature
+# determine how much material is melting or freezing). 
+# Because of that, we use a nonlinear solver scheme ('iterated IMPES')
+# that iterates between all of these equations, and we have to set its 
+# solver tolerance and the maximum number of iterations separately from 
+# the linear solver parameters. 
+set Nonlinear solver scheme                = iterated IMPES
+set Max nonlinear iterations               = 20
+set Nonlinear solver tolerance             = 1e-5
+set Number of cheap Stokes solver steps    = 100
+
+# In addition, melting and freezing normally happens on a much faster
+# time scale than the flow of melt, so we want to decouple the advection
+# of melt (and temperature) and the melting process itself. To do that, 
+# we use the operator splitting scheme, and define that for every 
+# advection time step, we want to do at least 50 reaction time step, 
+# and if these time steps would be larger than 100,000 years, we will do 
```

Comment:
This is a bit confusing, why not just write: "..., and we limit the maximum reaction time step length to ...".